pocs/camera/fli.py

```diff
@@ -291,9 +291,8 @@ def _readout(self, filename, width, height, header, exposure_event):
                 warn(message)
                 break
 
-        self._exposure_lock.release()
-
         fits_utils.write_fits(image_data, header, filename, self.logger, exposure_event)
+        self._exposure_lock.release()
```

Comment:
So by putting this after, will this just write out whatever data you have up to this point?

Reply:
Yes, I want the call the `write_fits` to happen even if the readout encounters errors occur because it's the writing of the FITS file to disc (or at least the attempt to) that results in the `exposure_event` being set.  Can't set the event before writing the FITS file because that would trigger `process_observation()` to try to process a file that doesn't exist yet.